diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java

```diff
@@ -1713,6 +1717,50 @@ private static void moveDir(final String source, final String destination) throw
         }
     }
 
+    /**
+     * Moves the contents from the source directory to the destination directory.
+     *
+     * @param sourceDir       The path to the source directory.
+     * @param destinationDir  The path to the destination directory.
+     * @throws IOException    If an I/O error occurs during the move operation.
+     */
+    private static void moveContents(final String sourceDir,
+                                     final String destinationDir)
+                                     throws IOException {
+        final Path sourcePath = Paths.get(sourceDir);
+        final Path destinationPath = Paths.get(destinationDir);
+
+        if (!Files.exists(sourcePath)) {
+            LOGGER.info("Source directory "
+                    + sourceDir
+                    + " does not exist. Skipping move operation.");
+            return;
+        }
+
+        // Create the destination directory if it doesn't exist
+        Files.createDirectories(destinationPath);
+
+        try (Stream<Path> stream = Files.list(sourcePath)) {
+            stream.forEach(path -> {
+                try {
+                    final Path targetPath = destinationPath.resolve(path.getFileName());
+                    if (Files.isDirectory(path)) {
+                        moveDir(path.toString(), targetPath.toString());
+                    }
+                    else {
+                        // Skip the JAR file
+                        if (!JAR_FILE_REGEX.equals(path.getFileName().toString())) {
```

Comment:
Please explain, if method is just to copy, it should copy all without consideration of some jar.

Reply:
The issue arose because the moveDir method is attempting to move or delete directories that contain files (like the JAR file) which should remain in place. This leads to a DirectoryNotEmptyException when trying to delete the source directory after moving its contents. After moving the files, we attempt to delete the source directory.
If it's not empty (due to the JAR file or other files we didn't move), we catch the `DirectoryNotEmptyException` and log a warning without throwing an error.

Error message upon running the workflow for DiffTool.java i.e. java-ci.yml:
```
[main] ERROR com.github.checkstyle.difftool.DiffTool - Error moving file reports/guava: tmp_reports/master/guava
[main] ERROR com.github.checkstyle.difftool.DiffTool - Error: reports
java.nio.file.DirectoryNotEmptyException: reports
```

Reply:
It is still hack.
Jar should not be placed in such folders that are for rename or remove.
We need to change another part of workflow or .... To not put jar in working folder of diff generation

Reply:
This existed earlier, with patch diff tool jar being placed in checkstyle-tester folder. Earlier diff.groovy would go onto the root i.e .contribution and access the src folder and compile and use the jar.

Reply:
the alternative is to bring in a new parameter for the java -jar execution that takes patch diff tool jar as args.

Reply:
Yes, a way better solution. We already wanted it in workflow

Reply:
To not put jar in working folder of diff generation: about this, when i tried to create the workflow for DifftTool.java, i came across this. Now, we possibly don't encounter this in the main repo as we use AWS.

Reply:
> Yes, a way better solution. We already wanted it in workflow

Okay.